Makefile

```diff
@@ -65,33 +71,87 @@ clean:
 	-@rm -f firedrake/cython/mg/impl.so > /dev/null 2>&1
 	@echo "    RM firedrake/cython/mg/impl.c"
 	-@rm -f firedrake/cython/mg/impl.c > /dev/null 2>&1
+	@echo "    RM tinyasm/*.so"
+	-@rm -f tinyasm/*.so
 
+# This is the minimum required to run the test suite
+NPROCS=8
+# On CI we add additional `pytest` args to spit out more information
+PYTEST_ARGS=
+# specifically --durations=200 --timeout=1800 --timeout-method=thread -o faulthandler_timeout=1860 -s
+# and --cov firedreake once the performance regression is fixed!
 
-THREADS=1
-ifeq ($(THREADS), 1)
-	PYTEST_ARGS=
-else
-	PYTEST_ARGS=-n $(THREADS)
-endif
+# Requires pytest and pytest-mpi only
+.PHONY: test_serial
+test_serial:
+	@echo "    Running all tests in serial"
+	@python -m pytest -v tests
+
+# Requires pytest and pytest-mpi only
+.PHONY: test_smoke
+test_smoke:
+	@echo "    Running the bare minimum smoke tests"
+	@python -m pytest -k "poisson_strong or stokes_mini or dg_advection" -v tests/regression/
```

Comment:
It would be better to use MPI on the "outside" here for the parallel tests so this can be run to check things on HPC